src/index.d.ts

```diff
@@ -0,0 +1,26 @@
+declare module "dwolla-v2" {
+  export class Client {
+    get: (
+      path: string,
+      query?: { [prop: string]: string },
+      headers?: { "Idempotency-Key"?: string } & { [prop: string]: string }
+    ) => Promise<Response>;
+    post: (
+      path: string,
+      body?: object,
+      headers?: { "Idempotency-Key"?: string } & { [prop: string]: string }
+    ) => Promise<Response>;
+    delete: (path: string) => Promise<Response>;
+    constructor(options: {
+      key: string;
+      secret: string;
+      environment?: "production" | "sandbox";
+    });
+  }
+
+  interface Response {
```

Comment:
Thanks for adding types to this project.

I would recommend supporting generics here and in all methods:
https://www.typescriptlang.org/docs/handbook/generics.html#hello-world-of-generics

E.g.:
```
interface Response<T=any> {
  status: number;
  headers: Headers;
  body: T;
```
This would then be easily used in the methods above too with something like:
```
get<T>: (
       path: string,
       query?: { [prop: string]: string },
       headers?: { "Idempotency-Key"?: string } & { [prop: string]: string }
     ) => Promise<Response<T>>;
```
allowing us to easily define the expected response in the body.

Reply:
Thanks @alias-mac. That sounds like a good idea to me! I just opened an issue: #57